src/test/java/edu/harvard/iq/dataverse/search/SolrSearchResultTest.java

```diff
@@ -3,11 +3,228 @@
 import javax.json.Json;
 import javax.json.JsonObject;
 import javax.json.JsonObjectBuilder;
+
 import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import org.junit.After;
+import org.junit.Before;
 import org.junit.Test;
 
+// ****************************************************************************************
+// The following tests test the setPublicationStatuses method aiming for 100% prime path coverage.
+// The following tool [1] was used to calculate the needed test paths.
+// 14 test paths (= 14 test methods) are needed.
+//
+// [1] https://cs.gmu.edu:8443/offutt/coverage/GraphCoverage
```

Comment:
Hi @matthew-a-dunlap, as you left a comment concerning the same issue on my PR too, I'll reply here.

Unfortunately, the tool does not convert source code into the graph that was posted above. This step needs to be done manually by looking at each statement of the source code and translating it to the graph (especially focusing on if statement and loops as they are the root of different execution paths of the method).

The tool then takes as input the:

- edges of the graph (e.g., `1 2`, `2 3`, `1 4`, ...)
- initial nodes of the graph (e.g., `1`)
- returning nodes of the graph (e.g., `3` and `7`)

_The examples of the list above refer to the graph posted by @nikzaugg._

As a result, the tool then generates test paths so that all prime paths are covered. These paths are then taken to derive test inputs to the method to fully cover all paths.

Reply:
Ah ok. Please remove the references to the graphing tool from the tests and write up a basic overview of the cases tested (its fine if it doesn't cover "all the edges of the graph"). I'm glad its been useful to understand and write tests cases, but its too much of a maintenance/code-rot burden to have info about it live on in the code.

Reply:
I removed the reference to the graphing tool in https://github.com/IQSS/dataverse/pull/5705/commits/1715ff7a02c2a28a68d4f7e26685fbffae17783e